gapic/src/main/com/google/gapid/perfetto/views/PowerSummaryPanel.java

```diff
@@ -0,0 +1,279 @@
+                ctx.fillPath(path);
+                ctx.drawPath(path);
+              });
+
+          if (hovered != null) {
+            double y = (trackHeight - 1) * (1 - (hovered.value - track.minValue) / range);
+            ctx.setBackgroundColor(mainGradient().highlight);
+            ctx.fillRect(hovered.startX, y - 1, hovered.endX - hovered.startX, trackHeight - y + 1);
+            ctx.setForegroundColor(colors().textMain);
+            ctx.drawCircle(hovered.mouseX, y, CURSOR_SIZE / 2);
+
+            ctx.setBackgroundColor(colors().hoverBackground);
+            double bgH = Math.max(hovered.size.h, trackHeight);
+            // The left x-axis coordinate of HoverCard.
+            double hx = hovered.mouseX + CURSOR_SIZE / 2 + HOVER_MARGIN;
+            if (hx >= w - (2 * HOVER_PADDING + hovered.size.w)) {
+              hx =
+                  hovered.mouseX
+                      - CURSOR_SIZE / 2
+                      - HOVER_MARGIN
+                      - 2 * HOVER_PADDING
+                      - hovered.size.w;
+            }
+            ctx.fillRect(
+                hx, Math.min((trackHeight - bgH) / 2, 0), 2 * HOVER_PADDING + hovered.size.w, bgH);
+            ctx.setForegroundColor(colors().textMain);
+            // The left x-axis coordinate of the left labels.
+            double x = hx + HOVER_PADDING;
+            y = (trackHeight - hovered.size.h) / 2;
+            // The difference between the x-axis coordinate of the left labels and the right labels.
+            double dx = hovered.leftWidth + HOVER_PADDING, dy = hovered.size.h / 2;
+            ctx.drawText(Fonts.Style.Normal, "Value:", x, y);
+            ctx.drawText(Fonts.Style.Normal, hovered.minLabel, x + dx, y);
+            ctx.drawText(Fonts.Style.Normal, hovered.maxLabel, x + dx, y + dy);
+
+            x = hx + HOVER_PADDING + hovered.leftWidth;
+            ctx.drawTextRightJustified(Fonts.Style.Normal, hovered.valueLabel, x, y, dy);
+
+            x = hx + HOVER_PADDING + hovered.size.w;
+            ctx.drawTextRightJustified(Fonts.Style.Normal, hovered.min, x, y, dy);
+            ctx.drawTextRightJustified(Fonts.Style.Normal, hovered.max, x, y + dy, dy);
+          }
+        });
+  }
+
+  @Override
+  protected Hover onTrackMouseMove(
+      Fonts.TextMeasurer m, Repainter repainter, double x, double y, int mods) {
+    PowerSummaryTrack.Data data = track.getData(state.toRequest(), onUiThread(repainter));
+    if (data == null || data.ts.length == 0) {
+      return Hover.NONE;
+    }
+
+    long time = state.pxToTime(x);
+    if (time < data.ts[0]) {
+      return Hover.NONE;
+    }
+
+    int idx = 0;
+    for (; idx < data.ts.length - 1; idx++) {
+      if (data.ts[idx + 1] > time) {
+        break;
+      }
+    }
+
+    if (idx >= data.ts.length) {
+      return Hover.NONE;
+    }
+
+    double startX = state.timeToPx(data.ts[idx]);
+    double endX = (idx >= data.ts.length - 1) ? startX : state.timeToPx(data.ts[idx + 1]);
```

Comment:
This could use a comment as to why `endX` is set to `startX`

Reply:
Will add.